torch/distributed/_spmd/experimental_ops.py

```diff
@@ -17,6 +17,38 @@
 aten = torch.ops.aten  # pyre-ignore
 
 
+@register_prop_rule(aten._foreach_add.List)  # pyre-ignore
```

Comment:
Hey @wanchaol is this specialization for foreach OK? If necessary, I can update `pointwise_rule` in a followup PR. It does not support list of tensors as args for now due to [this](https://github.com/pytorch/pytorch/blob/f388bec98583a31407526050900873ebe7463483/torch/distributed/_tensor/op_schema.py#L88)? 

https://github.com/pytorch/pytorch/blob/f388bec98583a31407526050900873ebe7463483/torch/distributed/_tensor/ops/common_rules.py#L248

Reply:
I think that's fine, we don't even need to update pointwise_rule as it might be non-trival to support list of inputs. I am planning on some op decomposition change that changes all of the rule registration, so things like `foreach_add` op is easier to express.